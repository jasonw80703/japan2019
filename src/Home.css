.home {
  text-align: center;
  margin: 150px;
}

.header {
  font-size: 150px;
}

.lucy {
  border: none;
}

.lucy-img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 75%;
  max-height: 75%;
}