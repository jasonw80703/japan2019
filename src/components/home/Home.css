.home {
  text-align: center;
  margin: 13%;
}

.header {
  font-size: 150px;
  font-weight: 300;
  color: #E85A4F;
}

.year {
  font-size: 140px;
}

.lucy {
  border: none;
}

.lucy-img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 75%;
  max-height: 75%;
}

.portrait-img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 60%;
  max-height: 60%;
}

#home-location-dropdown {
  margin-right: 20px;
  color: white;
  background-color: #E98074;
  border: none;
}

#misc-dropdown {
  color: #E98074;
  background-color: #EAE7DC;
  border-color: #E98074;
}

#misc-dropdown:hover {
  color: white;
  background-color: #E98074;
  border-color: #E98074;
}
