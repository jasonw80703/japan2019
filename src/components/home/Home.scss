.home {
  /* text-align: center; */
  margin: 13%;
}

.home .dropdown-menu {
  background-color: #E98074;
}

.home .dropdown-menu a {
  color: white;
}

.home .dropdown-menu a:hover {
  color: #E98074;
}

.header {
  font-size: 150px;
  font-weight: 300;
  color: #E85A4F;
}

.year {
  font-size: 140px;
}

.hideme {
  border: none;
}

.portrait-img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 60%;
  max-height: 60%;
}

#home-location-dropdown {
  margin-left: 30px;
  margin-right: 20px;
  color: white;
  background-color: #E98074;
  border: none;
}

#misc-dropdown {
  color: #E98074;
  background-color: #EAE7DC;
  border-color: #E98074;
}

#misc-dropdown:hover {
  color: white;
  background-color: #E98074;
  border-color: #E98074;
}
