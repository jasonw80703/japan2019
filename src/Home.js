import React, { Component } from 'react';
import {
  Dropdown,
  DropdownButton,
} from 'react-bootstrap';
import Modal from 'react-bootstrap/Modal';
import { LOCATIONS } from './constants/locations';
import lowerCaseRemoveSpaces from './helpers/lowerCaseRemoveSpaces';
import './Home.css';

import homeText from './assets/texts/home.json';
import lucy from './assets/images/home/lucy.jpg';

/**
 * Map locations constant to Dropdown with an Item for each location
 */
export function LocationDropdown() {
  return (
    <DropdownButton id="location-dropdown" variant="dark" title="Location" size="lg">
      {
        Object.values(LOCATIONS).map((location) => (
          <Dropdown.Item href={lowerCaseRemoveSpaces(location)} key={location}>{location}</Dropdown.Item>
        ))
      }
    </DropdownButton>
  );
};

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showLucy: false,
      showModal: false,
    };

    this.handleToggleModal = this.handleToggleModal.bind(this);
    this.handleToggleLucy = this.handleToggleLucy.bind(this);
  }

  handleToggleModal() {
    const { showModal } = this.state;

    this.setState({
      showModal: !showModal,
    });
  }

  handleToggleLucy() {
    this.setState({
      showLucy: !this.state.showLucy,
    })
  }

  render() {
    const { showLucy, showModal } = this.state;

    return (
      <div>
        <button type="button" className="lucy" onClick={this.handleToggleLucy}/>
        <div className="home">
          <h1 className="header" onClick={this.handleToggleModal}>{homeText.header}</h1>
          <LocationDropdown />
        </div>
        {showModal && (
          <Modal show={showModal} onHide={this.handleToggleModal}>
            <Modal.Header closeButton>
              <Modal.Title>About</Modal.Title>
            </Modal.Header>
            <Modal.Body>{homeText.modal.about}</Modal.Body>
            <Modal.Body>{homeText.modal.camera}</Modal.Body>
            <Modal.Body>{homeText.modal.howTo}</Modal.Body>
          </Modal>
        )}
        {showLucy && (
          <img className="lucy-img" src={lucy} alt="lucy" />
        )}
      </div>
    );
  }
}
