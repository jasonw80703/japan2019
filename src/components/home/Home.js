import React, { Component } from 'react';
import {
  Dropdown,
  DropdownButton,
  ButtonGroup,
} from 'react-bootstrap';
import Modal from 'react-bootstrap/Modal';
import { LOCATIONS } from '../../constants/locations';
import './Home.css';

import homeText from '../../assets/texts/home.json';
import lucy from '../../assets/images/home/lucy.jpg';
import portrait from '../../assets/images/home/portrait.jpg';

/**
 * Map locations constant to Dropdown with an Item for each location
 */
export function LocationDropdown() {
  return (
    <DropdownButton id="location-dropdown" variant="dark" title="Location" size="lg" as={ButtonGroup}>
      {
        Object.values(LOCATIONS).map((location) => (
          <Dropdown.Item href={location['href']} key={location['en']}>{location['en']}</Dropdown.Item>
        ))
      }
    </DropdownButton>
  );
};

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showLucy: false,
      showModal: false,
    };

    this.handleToggleModal = this.handleToggleModal.bind(this);
    this.handleToggleLucy = this.handleToggleLucy.bind(this);
  }

  handleToggleModal() {
    const { showModal } = this.state;

    this.setState({
      showModal: !showModal,
    });
  }

  handleToggleLucy() {
    const { showLucy } = this.state;

    this.setState({
      showLucy: !showLucy,
    })
  }

  render() {
    const { showLucy, showModal } = this.state;

    return (
      <div>
        <button type="button" className="lucy" onClick={this.handleToggleLucy}/>
        <div className="home">
          <h1 className="header" onClick={this.handleToggleModal}>{homeText.header}</h1>
          <LocationDropdown />
          <DropdownButton id="misc-dropdown" variant="light" title="Misc" size="lg" as={ButtonGroup}>
            <Dropdown.Item>Buildings</Dropdown.Item>
            <Dropdown.Item>Stamp Collection</Dropdown.Item>
            <Dropdown.Item>Music</Dropdown.Item>
            <Dropdown.Item>Coffee Shop Tier List</Dropdown.Item>
            <Dropdown.Item>Food Pictures</Dropdown.Item>
            <Dropdown.Item>Japan Travel Tips</Dropdown.Item>
          </DropdownButton>
        </div>
        {showModal && (
          <Modal show={showModal} onHide={this.handleToggleModal}>
            <Modal.Header closeButton>
              <Modal.Title>About</Modal.Title>
            </Modal.Header>
            <Modal.Body><img src={portrait} className="portrait-img" alt="portrait"/></Modal.Body>
            <Modal.Body>{homeText.modal.about}</Modal.Body>
            <Modal.Body>{homeText.modal.camera}</Modal.Body>
            <Modal.Body>{homeText.modal.howTo}</Modal.Body>
          </Modal>
        )}
        {showLucy && (
          <img className="lucy-img" src={lucy} alt="lucy" />
        )}
      </div>
    );
  }
}
