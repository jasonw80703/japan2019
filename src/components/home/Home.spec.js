import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Home from 'components/home/Home';

describe('<Home />', () => {
  let wrapper;

  it('renders as expected', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when handleToggleModal() is called', () => {
    wrapper = shallow(<Home />);
    wrapper.instance().handleToggleModal();
    expect(wrapper.state('showModal')).toEqual(true);
    wrapper.instance().handleToggleModal();
    expect(wrapper.state('showModal')).toEqual(false);
  });
});
