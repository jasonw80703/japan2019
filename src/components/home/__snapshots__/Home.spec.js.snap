// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders as expected 1`] = `
<div
  className="home"
>
  <h1
    className="header"
    onClick={[Function]}
  >
    日本
     
    <span
      className="year"
    >
      2019
    </span>
  </h1>
  <div
    className="dropdown btn-group"
    onKeyDown={[Function]}
    role="group"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-primary btn-lg"
      disabled={false}
      id="home-location-dropdown"
      onClick={[Function]}
      type="button"
    >
      Location
    </button>
  </div>
  <div
    className="dropdown btn-group"
    onKeyDown={[Function]}
    role="group"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-primary btn-lg"
      disabled={false}
      id="misc-dropdown"
      onClick={[Function]}
      type="button"
    >
      Misc
    </button>
  </div>
</div>
`;
